firedrake/interpolation.py

```diff
@@ -263,6 +314,18 @@ def _interpolator(V, tensor, expr, subset, arguments, access):
         # Replace with real coords coefficient
         coefficients[0] = coords
 
+    if runtime_vertex_only_mesh_interp:
+        # NOTE: Shouldn't this happen in compile_expression_dual_evaluation?
```

Comment:
Change this to comment what I'm actually doing rather than begging a question

Reply:
Also presumably `compile_expression_dual_evaluation` ought to be able to indicate whether it expects runtime arguments

Reply:
We should do that while redefining the surface interface so that we can cache kernels from `compile_expression...`.

Reply:
What do you mean by "the surface interface"?

Reply:
Just the API offered by the return value from `c_e_d_e`.